runbot_send_email/tests/test_runbot_send_email.py

```diff
@@ -0,0 +1,102 @@
+        branch_name = 'refs/heads/fast-travis'
+        self.repo_obj = self.env['runbot.repo'].create({
+            'name': repo_name,
+            'modules_auto': 'repo',
+            'mode': 'poll',
+            'nginx': True,
+        })
+        self.branch_obj = self.env['runbot.branch'].create({
+            'repo_id': self.repo_obj.id,
+            'name': branch_name,
+            'branch_name': 'fast-travis',
+            'sticky': False,
+            'coverage': False,
+        })
+        recipient = 'testcommitter@runbot.vauxoo.com'
+        self.build_obj = self.env['runbot.build'].create({
+            'branch_id': self.branch_obj.id,
+            'name': 'fcb98eb195fc62fa49873f8f101f1738e38ea7c0',
+            'author': 'Test Author',
+            'author_email': 'author@runbot.vauxoo.com',
+            'committer': 'Test Committer',
+            'committer_email': recipient,
+            'subject': '[TEST] Test message commit',
+            'date': '2016-03-08 00:00:00',
+       })
+        domain = [ ('repo_id','=',self.repo_obj.id), ]
+        self.build = self.build_obj.search(domain)
+        if os.environ.get('EMAIL_PASSWORD', False):
+            setattr(threading.currentThread(), 'testing', False)
+            self.build.pool._init = False
+            self.mail_server = self.env['ir.mail_server'].create({
+                'smtp_port': '25',
+                'smtp_host': 'smtp.mailgun.org',
+                'smtp_encryption': 'starttls',
+                'smtp_user': 'postmaster@runbot.vauxoo.com',
+                'smtp_pass': os.environ.get('EMAIL_PASSWORD'),
+                'name': 'test smtp server',
+                'sequence': 0,
+            })
+            # Added partner to set notify_email to always
+            partner_obj = self.env['res.partner']
+            partner_id = partner_obj.find_or_create(recipient)
+            partner = partner_obj.browse(partner_id)
+            partner.write({'notify_email': 'always'})
+
+    def tearDown(self):
+        """
+        This method is overwrite because
+        we need to reset values of setUp methods
+        """
+        super(TestRunbotSendEmail, self).tearDown()
+        setattr(threading.currentThread(), 'testing', True)
+        self.build.pool._init = True
+
+    def set_mail_password(self, password):
+        self.mail_server.write({
+            'smtp_pass': password,
+        })
+
+    def test_10_test_connection(self):
+        """
+        This makes the test smtp conection if EMAIL_PASSWORD env is set
+        """
+        if os.environ.get('EMAIL_PASSWORD'):
+            with self.assertRaisesRegexp( exceptions.except_orm,
+                                         'Connection Test Succeeded!'):
+                self.mail_server.test_smtp_connection()
+
+    def test_20_send_email(self):
+        """
+        This test tries to send a runbot email with password
+        """
+        if os.environ.get('EMAIL_PASSWORD'):
```

Comment:
We should get the tests without a password...
Check the main example: https://github.com/Vauxoo/yoytec/blob/8.0/mail_notification_collection/tests/test_template_collections.py#L74-L78